src/dialects/builtin/op_interfaces.rs

```diff
@@ -37,19 +38,31 @@ pub trait RegionKindInterface: Op {
 /// [Op]s that have exactly one region.
 pub trait OneRegionInterface: Op {
     fn get_region(&self, ctx: &Context) -> Ptr<Region> {
-        *self
-            .get_operation()
+        self.get_operation()
             .deref(ctx)
-            .regions
-            .get(0)
+            .get_region(0)
             .expect("Expected OneRegion Op to contain a region")
     }
+
+    /// Checks that the operation has exactly one region.
+    fn verify_one_region(&self, ctx: &Context) -> Result<(), CompilerError> {
```

Comment:
I took the liberty to extract one region check from `ModuleOp` and put it here to reuse in `FuncOp`. Would moving it to a separate trait be a better fit?

Reply:
I think what you have done is fine, for now.

Eventually, what we need is to automatically call interface verifiers. I don't know yet how to implement that in Rust. MLIR uses variadic template magic to do that automatically.

Reply:
Yeah, it'd be great to generate verification implementation from interfaces. I have high hopes for macros.